src/collections/Table/Table.js

```diff
@@ -10,43 +10,97 @@ export default class Table extends Component {
     children: ofComponentTypes(['TableColumn']),
     className: PropTypes.string,
     data: PropTypes.array,
+    onSelectRow: PropTypes.func,
     onSortChange: PropTypes.func,
+    selectedRows: PropTypes.arrayOf(PropTypes.number),
     sort: PropTypes.shape({
       key: PropTypes.string,
-      direction: PropTypes.string, // descending|ascending, defaults to descending
+      direction: PropTypes.oneOf(['descending', 'ascending']),
     }),
   };
 
+  static defaultProps = {
+    selectedRows: [],
+    sort: {
+      key: null,
+      direction: 'descending',
+    },
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      selectedRows: this.props.selectedRows,
+    };
+  }
+
   static getSafeCellContents(content) {
     // React cannot render objects, stringify them instead
     return _.isObject(content) ? JSON.stringify(content) : content;
   }
 
+  _isRowSelected(index) {
+    return _.includes(this.state.selectedRows, index);
+  }
+
+  _isSelectable = () => {
+    return _.includes(this.props.className, 'selectable');
+  };
```

Comment:
Since classes are (for now) the single source of truth for component options, this is used here.  The class is auto applied if any of the "*select*" props are used.  See render method.
>Note, in the future we plan to get away from use of Semantic UI classes in our components.  They will be replaced by props.